src/components/Layout.jsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from 'react'
+import Displayer from './Displayer'
+import NumControl from './NumControl'
+import Operators from './Operators'
+
+const Layout = () => {
+  const [operation, setOperation] = useState([])
+  const [result, setResult] = useState()
+  const [latestValue, setLatestValue] = useState()
+
+  const onReset = () => {
+    setOperation([])
+    setResult(null)
+    setLatestValue(null)
+  }
+
+  const getResult = () => {
+    const operationString = [...operation, latestValue].join(' ');
+    // eslint-disable-next-line no-new-func
+    const operationFunc = new Function(`return ${operationString}`);
```

Comment:
you can better wrap this in try catch block to safely process the unexpected error that can happen like when user put unsupported expression